sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/translation/ExpressionConverter.java

```diff
@@ -962,20 +980,28 @@ private RexNode convertResolvedParameter(ResolvedParameter parameter) {
   }
 
   private RexNode convertResolvedStructFieldAccess(ResolvedGetStructField resolvedGetStructField) {
-    return rexBuilder()
-        .makeFieldAccess(
-            convertRexNodeFromResolvedExpr(resolvedGetStructField.getExpr()),
-            (int) resolvedGetStructField.getFieldIdx());
+    RexNode referencedExpr = convertRexNodeFromResolvedExpr(resolvedGetStructField.getExpr());
+    return convertResolvedStructFieldAccessInternal(
```

Comment:
I have lost some of my memory on this code path:

just a question: does this work with deep nested ROW? E.g. `ROW(ROW(ROW(...)..)...)`

Reply:
Yes, see test I added (`testParameterStructNested`)